driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTest.java

```diff
@@ -123,6 +125,8 @@ public Entities getEntities() {
 
     protected abstract MongoClient createMongoClient(MongoClientSettings settings);
 
+    protected abstract GridFSBucket createGridFSBucket(MongoDatabase database);
```

Comment:
This is the abstraction that makes it possible to create an alternate implementation of GridFSBucket, similar to what we already do for creating a MongoClient.